### ISSUE TITLE
The latest Applovin Mediation 6.12 freeze iOS

### ISSUE THREAD

**Issue by neobie** [googleads/googleads-mobile-ios-mediation#214]
The latest Applovin Mediation 6.12.0 freeze iOS app by chance on start up (20-30%).
Reported on iPhone 8 and above with iOS 13.3 / 13.4

Tested with [ALSdk initializeSdk]; the chance is even higher (50% ?).

I can only test with Simulator iPhone 11 Max iOS 13.4.1.
After downgrading to 6.11.5 or removing mediation from Pod, there is no freeze.

How to test: close app from recent apps, and open app again, repeatedly.

**Comment by smartestapps** [comment]
I can't duplicate the bug using the same adapters, latest SDKs (don't know which versions you have) and device/iOS versions as you.

Between this and your other open issue about MoPub adapter crashing your project, I think there might be something else wrong.

Are you only testing on Simulator devices? If so, it could be that the computer you're working on is not powerful enough to run it, or it's set up incorrectly, or the Simulator is buggy, or a number of reasons.

If it also crashes on a real device, then perhaps look into your own code to see what is crashing it. If this is the case, try running the same SDK and adapters on a different project (perhaps one that's much simpler). From there, you'll get a better idea of what the cause of the crash is.

**Comment by thomasmso** [comment]
@neobie - do you mind emailing me at thomas.so@applovin.com? I can take care of you. Thanks.

**Comment by neobie** [comment]
@smartestapps I tested with my iPad Air 2 , same freeze on startup happen. Only if I downgraded to 6.11.5, so far no more freeze. I cant get any freeze / crash detail from Firebase / Crashlytics. Not related to Simulator. My Macbook is 2017 Macbook pro 15' with highest spec 16GB Ram Model.

@thomasmso I am releasing new version to users, let's see if the issue resolved.

**Comment by neobie** [comment]
@smartestapps Who is in charge for the Admob adapter? Have you guys tested if each adapters actually works with each other by installing all of them one 1 app ? Since they are 10~20 mediation in Admob?

**Comment by smartestapps** [comment]
@neobie I don't work for Google. I'm just another developer who also uses these adapters trying to help out a fellow developer. I do have the newest Applovin and MoPub adapters in my app, and they haven't caused any crashes/freezes. Your issues reporting had me worried that I was putting extremely buggy code into my highly popular app, so it caught my attention. After thorough testing with iOS 13.3-13.4.1, I couldn't get the app to crash like you said it would.

Have you tried creating a simple hello world project and adding the SDKs and adapters there? Are you using the newest SDKs (Admob and Applovin)?

If you have done that and it still crashes, you could post that code online and maybe someone here will figure out the issue for you. Because with no crash log and no code to see, I don't think anyone can figure out what the issue is when the adapter doesn't crash on their projects.

**Comment by ericleich** [comment]
@neobie +1 to smarttestapps suggestion above, it's hard to debug without some sort of crash log or console logs.

**Comment by neobie** [comment]
Hi, I wish to inform that after downgrading AppLovin to 6.11.5 (and no other SDK or coding changes), no app freeze happens, and so far all users feedback are positive.

I did save pod update logs to back track any issues. 

Here is the pod log before my app get freeze:

Further, I did test by disabling all mediations, leaving Applovin 6.12. The app still freeze by chance (1/10). Putting [ALSdk initializeSdk] in AppDelegate.m increase the chance to freeze (5/10).

I will post again if I manage to test with Hello world.
